src/utils/memoization.ts

```diff
@@ -29,7 +29,7 @@ export function createMemoizationKeyFromCalendarOptionsAndYear(
 function stringifyCalendarOptions(
   retailCalendarOptions: RetailCalendarOptions,
 ) {
-  return Object.entries(retailCalendarOptions)
-    .map((e) => e.join(':'))
-    .join('-')
+  const { weekGrouping, lastDayOfWeek, lastMonthOfYear, weekCalculation, addLeapWeekToMonth, beginningMonthIndex, customLeapYearOptions } = retailCalendarOptions
+  return `1:${weekGrouping}-2:${lastDayOfWeek}-3:${lastMonthOfYear}-4:${weekCalculation}-5:${addLeapWeekToMonth}-6:${beginningMonthIndex}-7:${customLeapYearOptions?.calendarYear}-8:${customLeapYearOptions?.yearEndDate}-9:${customLeapYearOptions?.leapYearFrequency}`
```

Comment:
@yakuza8 I am conflicted about this change. Now that we have multi-level RetailCalendarOptions, I am afraid that a generic recursive solution that traverses options object might be too slow. 

This is not future-proof, as adding new fields will break key generaration.
I am open to ideas on how to approach this.

Reply:
Thanks for the suggestion! I agree that such a lightweight dependency would be suitable option.
For now, I will keep options object flat, it also makes it much easier to map to a db table on our backend.